File: admin/src/types/document.ts
export type DocumentRootRequest = {
  title: string;
  content: string;
  description: string;
  thumbnail: string;
  document_type: string;
  is_public: boolean;
};

export type DocumentRequest = {
  title: string;
  content: string;
  document_type: string;
  is_public: boolean;
  parent_id: string;
  document_id: string;
};

export type DocumentSaveRequest = {
  id: string;
  title: string;
  content: string;
};

export type DocumentVO = {
  id: string;
  created_at: string;
  updated_at: string;
  title: string;
  content: string;
  document_type: string;
  is_public: boolean;
  parent_id: string;
  document_id: string;
};

export type DocumentRootVO = {
  id: string;
  created_at: string;
  updated_at: string;
  title: string;
  description: string;
  thumbnail: string;
  is_public: boolean;
};

export type DocumentTreeVO = {
  id: string;
  created_at: string;
  updated_at: string;
  title: string;
  document_type: string;
  is_public: boolean;
  parent_id: string;
  document_id: string;
};

File: server/internal/document/internal/domain/document.go
package domain

import (
	"time"

	"go.mongodb.org/mongo-driver/v2/bson"
)

type Document struct {
	ID         bson.ObjectID
	CreatedAt  time.Time
	UpdatedAt  time.Time
	Title      string
	Content    string
	Description  string
	Thumbnail    string
	DocumentType string
	IsPublic     bool
	ParentID     bson.ObjectID
	DocumentID   bson.ObjectID
}
